src/plugins/data_source_management/public/components/direct_query_data_sources_components/connection_detail/access_control_tab.tsx

```diff
@@ -34,17 +35,17 @@ export const AccessControlTab = (props: AccessControlTabProps) => {
   const { http } = useOpenSearchDashboards<DataSourceManagementContext>().services;
 
   useEffect(() => {
-    http!
-      .get(SECURITY_ROLES)
-      .then((data) =>
+    http
+      .get(SECURITY_ROLES, { query: { dataSourceId: props.dataSourceMDSId } })
```

Comment:
There should ideally be some constant that core exposes that plugins can pull from, along same idea as https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6534. That way, this is not a hard coded value that can be changed/broken in the future

Reply:
Yes, I was thinking the same and we should do this as a followup